File: src/main/java/it/contrader/wmesspring/model/TaskModel.java
package it.contrader.wmesspring.model;

import java.io.Serializable;
import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * The persistent class for the tasks_model database table.
 * 
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "tasks_model")
public class TaskModel implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "task_model_id")
	private int taskModelId;

	@Column(name = "task_model_action")
	private String taskModelAction;

	@Column(name = "task_model_description")
	private String taskModelDescription;

	@Column(name = "task_model_state")
	private int taskModelState;

	@Column(name = "task_model_time")
	private int taskModelTime;

	// bi-directional many-to-many association to Item
	@ManyToMany
	@JoinTable(name = "tasks_model_items", joinColumns = { @JoinColumn(name = "task_model_id") }, inverseJoinColumns = {
			@JoinColumn(name = "item_id") })
	private List<Item> items;

	// bi-directional many-to-one association to Resource
	@ManyToOne
	@JoinColumn(name = "resource_id")
	private Resource resource;

	// bi-directional many-to-one association to User
	@ManyToOne
	@JoinColumn(name = "user_id")
	private User user;

}

File: src/app/models/TaskModel.ts
import { User } from './User';
import { Item } from './Item';
import { Resource } from './Resource';


export class TaskModel {
    taskModelId: number;
    resourceDTO: Resource;
    taskModelAction: string;
    taskModelDescription: string;
    taskModelState: number;
    taskModelTime: number;
    userDTO: User;
    itemsDTO: Item[];

    // tslint:disable-next-line:max-line-length
    constructor(taskModelId: number, taskModelAction: string, taskModelDescription: string, taskModelState: number, taskModelTime: number, userDTO: User, resourceDTO: Resource, itemsDTO: Item[]) {

        this.taskModelId = taskModelId;
        this.taskModelAction = taskModelAction;
        this.taskModelDescription = taskModelDescription;
        this.taskModelState = taskModelState;
        this.taskModelTime = taskModelTime;
        this.userDTO = userDTO;
        this.resourceDTO = resourceDTO;
        this.itemsDTO = itemsDTO;
        /*
        this.projectsDTO = projectsDTO;
        this.projectTemplatesDTO = projectTemplatesDTO;
        */
    }
}

